service/evidence/evidence_store.go

```diff
@@ -36,63 +36,66 @@ func init() {
 }
 
 // StoreEvidence is a method implementation of the evidenceServer interface: It receives an req and stores it
-func (s *Service) StoreEvidence(_ context.Context, req *evidence.StoreEvidenceRequest) (*evidence.StoreEvidenceResponse, error) {
-	var (
-		resp = &evidence.StoreEvidenceResponse{}
-		err  error
-		e    = req.GetEvidence()
-	)
+func (s *Service) StoreEvidence(_ context.Context, req *evidence.StoreEvidenceRequest) (resp *evidence.StoreEvidenceResponse, err error) {
 
-	_, err = e.Validate()
-	if err != nil {
-		log.Errorf("Invalid evidence: %v", err)
-		newError := fmt.Errorf("invalid evidence: %w", err)
+	err = s.handleEvidence(req.GetEvidence())
 
-		// Inform our hook, if we have any
-		if s.EvidenceHook != nil {
-			for _, hook := range s.EvidenceHook {
-				go hook(nil, newError)
-			}
+	if err != nil {
+		resp = &evidence.StoreEvidenceResponse{
+			Status: false,
 		}
 
-		return resp, status.Errorf(codes.InvalidArgument, "invalid req: %v", err)
+		return resp, status.Errorf(codes.Internal, "Error while handling evidence: %v", err)
```

Comment:
The problem that I see here is that it is now indistinguishable whether we have an internal error (such as the DB connection was lost or so), which the user cannot fix from an invalid argument error, which the user can fix, e.g. by supplying the correct argument.

Reply:
That's a good point.

Reply:
Done in PR #430.